testing/sample_framework/src/android/android_app_framework.cc

```diff
@@ -478,6 +479,43 @@ std::string ReadTextInput(const char* title, const char* message,
   return g_text_entry_field_data->ReadText(title, message, placeholder);
 }
 
+void SetEnvironmentVariableFromExtra(const char* extra_name, JNIEnv* env,
+                                     jobject intent,
+                                     jmethodID get_string_extra) {
+  jstring extra_value_jstring = (jstring)env->CallObjectMethod(
+      intent, get_string_extra, env->NewStringUTF(extra_name));
+
+  if (extra_value_jstring != nullptr) {
+    const char* extra_value =
+        env->GetStringUTFChars(extra_value_jstring, nullptr);
+    setenv(extra_name, extra_value == nullptr ? "" : extra_value,
+           /*overwrite=*/1);
+    env->ReleaseStringUTFChars(extra_value_jstring, extra_value);
+    env->DeleteLocalRef(extra_value_jstring);
+  }
+}
+
+void SetExtrasAsEnvironmentVariables() {
+  JNIEnv* env = app_framework::GetJniEnv();
+  jobject activity = g_app_state->activity->clazz;
```

Comment:
I think that `g_app_state->activity->clazz` can be replaced by `app_framework::GetActivity()`.

Reply:
Done.